### PR TITLE
Refs [TASK][38392] Create UI, Handle events AddPassengerScreen

### PR DESCRIPTION
**Pull request by vinhmone**
## Related Tickets
- [#Task 38392: Create UI AddPassengerScreen](https://edu-redmine.sun-asterisk.vn/issues/38392)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import androidx.core.os.bundleOf
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.sun.findflight.databinding.FragmentAddPassengerBinding
+import com.sun.findflight.ui.home.HomeFragment
+import com.sun.findflight.utils.showToast
+
+class AddPassengerFragment : BottomSheetDialogFragment(), View.OnClickListener {
+
+    private var _viewBinding: FragmentAddPassengerBinding? = null
+    private lateinit var viewBinding: FragmentAddPassengerBinding
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt:19]: sao k cho vào base nhỉ

File: app/src/main/res/layout/fragment_add_passenger.xml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        app:layout_constraintBottom_toTopOf="@id/textAgeRangeAdult"
+        app:layout_constraintStart_toStartOf="@id/imageDividerAdult"
+        app:layout_constraintTop_toTopOf="@id/imageAddAdult" />
+
+    <TextView
+        android:id="@+id/textAgeRangeAdult"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/text_age_adult"
+        android:textSize="14sp"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_add_passenger.xml]: hard code

File: app/src/main/res/layout/fragment_add_passenger.xml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        app:layout_constraintBottom_toTopOf="@id/textAgeRangeChild"
+        app:layout_constraintStart_toStartOf="@id/imageDividerChild"
+        app:layout_constraintTop_toBottomOf="@id/imageDividerChild" />
+
+    <TextView
+        android:id="@+id/textAgeRangeChild"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/text_age_child"
+        android:textSize="14sp"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_add_passenger.xml]: .

File: app/src/main/res/layout/fragment_add_passenger.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        app:layout_constraintStart_toStartOf="@id/imageDividerInfant"
+        app:layout_constraintTop_toBottomOf="@id/imageDividerInfant" />
+
+    <TextView
+        android:id="@+id/textAgeRangeInfant"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:contentDescription="@null"
+        android:text="@string/text_age_infant"
+        android:textSize="14sp"
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_add_passenger.xml]: .

File: app/src/main/res/layout/fragment_add_passenger.xml (hunk 3 above)
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_add_passenger.xml]: .

File: app/src/main/res/layout/fragment_add_passenger.xml [hunk 5]
```diff
@@ -0,0 +1,266 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
```
Comment (thonc-2135) [inline app/src/main/res/layout/fragment_add_passenger.xml]: ????

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        _viewBinding = null
+    }
+
+    companion object {
+        const val MESSAGE_OVER_PEOPLE_NUMBER = "Max 9 passengers!"
+        const val MESSAGE_BELOW_ADULT_NUMBER = "At least 1 adult!"
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt]: sao k cho vào string mà ở đây v nhỉ

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    private fun initEvents() {
+        viewBinding.run {
+            listOf(
+                imageMinusAdult, imageAddAdult,
+                imageAddChild, imageMinusChild,
+                imageAddInfant, imageMinusInfant,
+                buttonApplyPassenger
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt]: style hay v

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    private var total = DEFAULT_TOTAL
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _viewBinding = FragmentAddPassengerBinding.inflate(inflater, container, false)
+        _viewBinding?.let { viewBinding = it }
+        return _viewBinding?.root
```
Comment (thonc-2135) [inline app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt:29]: cho vào base nốt đi

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt (hunk 1 above)
Comment (vinhmone) [reply app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt:19]: cái này nó extend từ BottomSheetDialogFragment thì cho vào base thế nào a

File: app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt (hunk 7 above)
Comment (vinhmone) [reply app/src/main/java/com/sun/findflight/ui/addpassenger/AddPassengerFragment.kt]: e chia nó theo từng cụm nút đó ạ, để e fix lại :D

**Pull request merged**